pkg/controller/bundlec/resource_sync_task.go

```diff
@@ -574,6 +575,12 @@ func (st *resourceSyncTask) updateResource(resClient dynamic.ResourceInterface,
 		return updated, false, nil
 	}
 
+	if difference == "" {
+		st.logger.Info("Objects are different: Secret object has changed")
```

Comment:
Given that difference is just a string, I would be tempted to return just "secret object difference not reported" from the CompareActualVsSpec function, and not have this slightly messy interpretation of an empty string.

Reply:
good point. How did it not occur to me...